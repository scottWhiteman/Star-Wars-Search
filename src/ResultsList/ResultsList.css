.Results-List {
    padding: 20px;
    margin: 0 50px;
}

.Results-Item {
    background-color: #333;
    border: 3px dotted #eaca00;
    border-radius: 25em;
    list-style: none;
    padding: 20px;
    margin: 20px;
}