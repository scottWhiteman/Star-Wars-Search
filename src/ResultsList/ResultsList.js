import React from 'react';
import './ResultsList.css';
import Result from '../Result/Result';

export default class ResultsList extends React.Component {
    static defaultProps = {
        results: []
    }
    
    renderList() {
        if (this.props.results.length <= 0) {
            return <h2>No results found!</h2>
        }
        return this.props.results.map((single, index) => {
            return <li key={index} className="Results-Item">
                    <Result
                        item={single}
                        category={this.props.category}/>
                </li>
        })
    }
    
    render() {
        return (
            <ul className="Results-List">
                {this.renderList()}
            </ul>  
        );
    }
}