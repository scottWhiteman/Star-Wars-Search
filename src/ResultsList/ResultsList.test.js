import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ResultsList from './ResultsList';

describe("Results List Component", () => {
    const props = {
        results: [
            {
                "name": "Luke",
                "species": "Test",
                "homeworld": "JavaScript"
            },
            {
                "name": "Darth",
                "species": "Helmet",
                "homeworld": "Space"
            },
            {
                "name": "Burrito",
                "species": "Food",
                "homeworld": "Fridge"
            }
        ]
    }

    it("renders without crashing", () => {
        const div = document.createElement('div');
        ReactDOM.render(<ResultsList />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    
    it("renders empty list", () => {
        const tree = renderer.create(<ResultsList results={[]}/>).toJSON()
        expect(tree).toMatchSnapshot()
    });

    it("renders each item result", () => {
        const tree = renderer.create(<ResultsList results={props.results}/>).toJSON()
        expect(tree).toMatchSnapshot()
    });
})