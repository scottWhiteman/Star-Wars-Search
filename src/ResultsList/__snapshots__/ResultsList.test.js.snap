// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results List Component renders each item result 1`] = `
<ul
  className="Results-List"
>
  <li
    className="Results-Item"
  >
    <div
      className="Result-Container"
    >
      <h2>
        That Category does not exist!
      </h2>
    </div>
  </li>
  <li
    className="Results-Item"
  >
    <div
      className="Result-Container"
    >
      <h2>
        That Category does not exist!
      </h2>
    </div>
  </li>
  <li
    className="Results-Item"
  >
    <div
      className="Result-Container"
    >
      <h2>
        That Category does not exist!
      </h2>
    </div>
  </li>
</ul>
`;

exports[`Results List Component renders empty list 1`] = `
<ul
  className="Results-List"
>
  <h2>
    No results found!
  </h2>
</ul>
`;

exports[`Results List Component renders without crashing 1`] = `
<ul
  className="Results-List"
>
  <h2>
    No results found!
  </h2>
</ul>
`;
